Week05/lesson10/homework/City Changer/js/index.js

```diff
@@ -0,0 +1,30 @@
+var city = document.getElementById("#city-type").value;
+
+function switchBackground(event){
+event.preventDefault();
+document.querySelector("body").className = "city-type";
```

Comment:
You're not going to want to set body's class to be the string `"city-type"` because then your html will look like this:

`<body class="city-type">`

What you want is the set the class of body to the value currently typed into the input (which has an ID of `"city-type"`. If only we had stored that value somewhere...

Oh wait! we do, in the corrected version of line 1:

`var city = document.querySelector("#city-type").value;`

So cool, instead what I want is something like this:

`document.querySelector("body").className = city;`

So I refresh the page and... huh nothing is happening when I type in "london", I can even see by opening up my Chrome Inspector that the class "london" isn't getting assigned to the body:

![image](https://cloud.githubusercontent.com/assets/5670279/17868170/9d484f02-68a5-11e6-968c-8c9d9269cfe9.png)

What gives? Let's look at the updated code:

```
var city = document.querySelector("#city-type").value;

function switchBackground(event){
event.preventDefault();
document.querySelector("body").className = city;

    if(city === "!==") {
      switchBackground("class", "home");
    }
    else if(city === "austin") {
      switchBackground("class", "austin");
    }
    else if(city === "la") {
      switchBackground("class", "la");
    }
    else if(city === "london") {
      switchBackground("class", "london");
    }
    else if(city === "new york city") {
      switchBackground("class", "nyc");
    }
    else if(city === "san francisco") {
      switchBackground("class", "sf");
    }
    else if(city === "sydney") {
      switchBackground("class", "sydney");
    }
  }

document.querySelector("#submit-btn").onclick = switchBackground
```

... well first and foremost, I can see that my indentation is a bit weird. My function starts flush to the left but the closing `}` is indented one level. Also the first two lines within my function aren't indented but they should be. Let's sort out my indentation:

```
var city = document.querySelector("#city-type").value;

function switchBackground(event){
  event.preventDefault();
  document.querySelector("body").className = city;

  if(city === "!==") {
    switchBackground("class", "home");
  }
  else if(city === "austin") {
    switchBackground("class", "austin");
  }
  else if(city === "la") {
    switchBackground("class", "la");
  }
  else if(city === "london") {
    switchBackground("class", "london");
  }
  else if(city === "new york city") {
    switchBackground("class", "nyc");
  }
  else if(city === "san francisco") {
    switchBackground("class", "sf");
  }
  else if(city === "sydney") {
    switchBackground("class", "sydney");
  }
}

document.querySelector("#submit-btn").onclick = switchBackground; /* adding forgotten semi-colon too */
```

Aha!! Why is the first line, where I get the value of `city` chilling outside the function by itself?? My `switchBackground` function is tied to a very specific moment in time (the `click`), and it's then and only then that I want to grab the value off of the input. As it stands now, I'm grabbing the value the minute the page loads, before the user has had time to fill in the input, so that value of `city` is always going to be nothing!

Let's move it inside the function:

```
function switchBackground(event){
  event.preventDefault();
  var city = document.querySelector("#city-type").value;
  document.querySelector("body").className = city;

  if(city === "!==") {
    switchBackground("class", "home");
  }
  else if(city === "austin") {
    switchBackground("class", "austin");
  }
  else if(city === "la") {
    switchBackground("class", "la");
  }
  else if(city === "london") {
    switchBackground("class", "london");
  }
  else if(city === "new york city") {
    switchBackground("class", "nyc");
  }
  else if(city === "san francisco") {
    switchBackground("class", "sf");
  }
  else if(city === "sydney") {
    switchBackground("class", "sydney");
  }
}

document.querySelector("#submit-btn").onclick = switchBackground;
```

Now I save, and refresh the page. I type in `london` into the input, hit enter and boom it works!

![image](https://cloud.githubusercontent.com/assets/5670279/17868524/0c010cf8-68a7-11e6-8da7-664fdda8b657.png)

That said, it seems like only the first few lines of my code are doing all the work... Is my `if/else` statement is working right let's `console.log` the inside the `if/else` statement to make sure it's working:

```
function switchBackground(event){
  event.preventDefault();
  var city = document.querySelector("#city-type").value;
  document.querySelector("body").className = city;

  if(city === "!==") {
    switchBackground("class", "home");
  }
  else if(city === "austin") {
    switchBackground("class", "austin");
  }
  else if(city === "la") {
    switchBackground("class", "la");
  }
  else if(city === "london") {
    console.log("inside london");
    switchBackground("class", "london");
  }
  else if(city === "new york city") {
    switchBackground("class", "nyc");
  }
  else if(city === "san francisco") {
    switchBackground("class", "sf");
  }
  else if(city === "sydney") {
    switchBackground("class", "sydney");
  }
}

document.querySelector("#submit-btn").onclick = switchBackground;
```

Now I refresh and type in `london` and keep my eye on the console as I hit the "Update" button...

![image](https://cloud.githubusercontent.com/assets/5670279/17868646/9b9d5d12-68a7-11e6-87df-accddd646a58.png)

Huh that's weird, I see "inside london" once and then an error... Let's try `console.log`'ing right before we use the `event`:

```
function switchBackground(event){
  console.log("hitting function");
  event.preventDefault();
  var city = document.querySelector("#city-type").value;
  document.querySelector("body").className = city;

  if(city === "!==") {
    switchBackground("class", "home");
  }
  else if(city === "austin") {
    switchBackground("class", "austin");
  }
  else if(city === "la") {
    switchBackground("class", "la");
  }
  else if(city === "london") {
    console.log("inside london");
    switchBackground("class", "london");
  }
  else if(city === "new york city") {
    switchBackground("class", "nyc");
  }
  else if(city === "san francisco") {
    switchBackground("class", "sf");
  }
  else if(city === "sydney") {
    switchBackground("class", "sydney");
  }
}

document.querySelector("#submit-btn").onclick = switchBackground;
```

I do the usual steps and I see:

![image](https://cloud.githubusercontent.com/assets/5670279/17868721/f49e5218-68a7-11e6-9061-0d392f0a9e6e.png)

Huh? Why am I seeing `hitting function` twice?

Upon looking at my code I finally see it! I calling the function `switchBackground` from *inside* `switchBackground` and that's why I'm seeing `hitting function` twice!!

I know I'm going to need a nice pretty new function instead, that will take care of the styling, so let's write one and then use that instead:

```
function makeItPretty(thingy) {
  document.querySelector("body").className = thingy;
}

function switchBackground(event){
  event.preventDefault();
  var city = document.querySelector("#city-type").value;

  if(city === "!==") {
    makeItPretty("home");
  }
  else if(city === "austin") {
    makeItPretty("austin");
  }
  else if(city === "la") {
    makeItPretty("la");
  }
  else if(city === "london") {
    makeItPretty("london");
  }
  else if(city === "new york city") {
    makeItPretty("nyc");
  }
  else if(city === "san francisco") {
    makeItPretty("sf");
  }
  else if(city === "sydney") {
    makeItPretty("sydney");
  }
}

document.querySelector("#submit-btn").onclick = switchBackground;
```

... and I think I'll leave you there Jo!! Take it away from here